src/app/clusters/service-area-server/service-area-delegate.cpp

```diff
@@ -0,0 +1,101 @@
+#include "service-area-delegate.h"
+#include "service-area-server.h"
+
+using namespace chip::app::Clusters::ServiceArea;
+
+bool Delegate::GetSupportedLocationById(uint32_t aLocationId, uint32_t & listIndex, LocationStructureWrapper & aSupportedLocation)
+{
+    bool ret_value = false;
+
+    listIndex = 0;
+
+    // simple linear iteration to find the location with the desired locationID.
+    while (GetSupportedLocationByIndex(listIndex, aSupportedLocation))
+    {
+        if (aSupportedLocation.locationID == aLocationId)
+        {
+            ret_value = true;
+            break;
+        }
+
+        ++listIndex;
+    }
+
+    return ret_value;
+};
+
+/**
+ * When we cannot guarantee that the restrictions imposed on the SupportedLocations, CurrentLocation and Progress attributes
+ * can be upheld, these attributes are set to null.
+ *
+ * The user is free the redefine this method as their device may have more information on what has changed and may be able to
+ * maintain the restrictions on these attributes by selectively editing them.
+ */
+void Delegate::HandleSupportedLocationsUpdated()
+{
+    mInstance->ClearSelectedLocations();
+    mInstance->SetCurrentLocation(DataModel::Nullable<uint32_t>());
+    mInstance->ClearProgress();
+};
+
+bool Delegate::GetSupportedMapById(uint8_t aMapId, uint32_t & listIndex, MapStructureWrapper & aSupportedMap)
+{
+    bool ret_value = false;
```

Comment:
update style throughout the PR: please early return for code clarity and do not use declaring the return variable at the top.

Reply:
These have been removed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/683b0fe4bf1ed901ea7d48a9c41fa5c59ccd921a).